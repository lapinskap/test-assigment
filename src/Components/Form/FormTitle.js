import PropTypes from 'prop-types';
import React from 'react';
import { CardHeader } from 'reactstrap';
import Sticky from 'react-stickynode';
import __ from '../../utils/Translations';

export default function FormTitle({
  stickyTitle, title, buttons, translateTitle,
}) {
  return (
    <Sticky enabled={stickyTitle} top=".app-header" activeClass="sticky-active-class">
      <CardHeader className="card-header-lg text-transform-none">
        <div className="card-header-title font-size-lg font-weight-normal">
          {translateTitle ? __(title) : title}
        </div>
        <div className="btn-actions-pane-right">
          {buttons}
        </div>
      </CardHeader>
    </Sticky>
  );
}

FormTitle.propTypes = {
  buttons: PropTypes.node,
  stickyTitle: PropTypes.bool,
  translateTitle: PropTypes.bool,
  title: PropTypes.oneOfType([PropTypes.string, PropTypes.node]),
};

FormTitle.defaultProps = {
  buttons: [],
  stickyTitle: false,
  translateTitle: true,
  title: '',
};
