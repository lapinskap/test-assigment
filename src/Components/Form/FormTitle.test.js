import React from 'react';
// import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import FormTitle from './FormTitle';
import '../../tests/setupTests';

function createNodeMock() {
  return {
    getBoundingClientRect() {
      return {
        width: 100,
      };
    },
  };
}

describe('<FormTitle />', () => {
  const setState = jest.fn();
  const useStateSpy = jest.spyOn(React, 'useState');
  useStateSpy.mockImplementation((init) => [init, setState]);
  let title;
  let validateForm;

  beforeEach(() => {
    title = 'Test title';
    validateForm = () => {
    };
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('renders correctly with all props', () => {
    const buttons = [
      <button type="submit">Button 1</button>,
      <button type="button">Button 2</button>,
      <button type="button">Button 3</button>,
    ];
    const tree = renderer
      .create(<FormTitle
        buttons={buttons}
        title={title}
        validateForm={validateForm}
      />, { createNodeMock })
      .toJSON();

    expect(tree)
      .toMatchSnapshot();
  });

  // it('Should capture title correctly onChange', () => {
  //   const formTitle = shallow(<FormTitle
  //     buttons={buttons}
  //     title={title}
  //     stickyTitle
  //     validateForm={validateForm}
  //   />).find('.card-header-title');
  //   formTitle.value = 'Test';
  //   formTitle.simulate('change');
  //   expect(setState).toHaveBeenCalledWith('Test');
  // });

  it('renders correctly with valid buttons', () => {
    const buttons = [
      <button type="submit">Button 1</button>,
      <button type="button">Button 2</button>,
      <button type="button">Button 3</button>,
    ];
    const tree = renderer
      .create(<FormTitle
        buttons={buttons}
        title={title}
        stickyTitle
        validateForm={validateForm}
      />, { createNodeMock })
      .toJSON();

    expect(tree)
      .toMatchSnapshot();
  });
});
