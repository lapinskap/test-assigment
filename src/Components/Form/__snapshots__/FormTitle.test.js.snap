// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FormTitle /> renders correctly with all props 1`] = `
<div
  className="sticky-outer-wrapper"
  style={Object {}}
>
  <div
    className="sticky-inner-wrapper"
    style={
      Object {
        "position": "relative",
        "top": "0px",
        "zIndex": undefined,
      }
    }
  >
    <div
      className="card-header-lg text-transform-none card-header"
    >
      <div
        className="card-header-title font-size-lg font-weight-normal"
      >
        Test title
      </div>
      <div
        className="btn-actions-pane-right"
      >
        <button
          type="submit"
        >
          Button 1
        </button>
        <button
          type="button"
        >
          Button 2
        </button>
        <button
          type="button"
        >
          Button 3
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`<FormTitle /> renders correctly with valid buttons 1`] = `
<div
  className="sticky-outer-wrapper"
  style={Object {}}
>
  <div
    className="sticky-inner-wrapper"
    style={
      Object {
        "position": "relative",
        "top": "",
        "transform": "translate3d(0,0px,0)",
        "zIndex": undefined,
      }
    }
  >
    <div
      className="card-header-lg text-transform-none card-header"
    >
      <div
        className="card-header-title font-size-lg font-weight-normal"
      >
        Test title
      </div>
      <div
        className="btn-actions-pane-right"
      >
        <button
          type="submit"
        >
          Button 1
        </button>
        <button
          type="button"
        >
          Button 2
        </button>
        <button
          type="button"
        >
          Button 3
        </button>
      </div>
    </div>
  </div>
</div>
`;
